### PR TITLE
More Optimus Events

### PR DESCRIPTION
**Pull request by Mryashbhardwaj**
Emit events on 

1. Replay : create
2. Job : Update and Delete
3. Job Run : Success, Failure and SLA miss
4. Resource : Update and Delete

### CODE REVIEW

File: core/job/service/job_service.go [hunk 1]
```diff
@@ -72,12 +75,18 @@ type JobService struct {
 	logger log.Logger
 }
 
+type AlertManager interface {
+	io.Closer
```
Comment (sbchaos) [inline core/job/service/job_service.go:79]: We are not using `Close()` function in this file. so we should not have this as part of contract.

**Review by irainia — COMMENTED** [review]
Just small feedbacks to consider.

File: core/job/spec.go [hunk 2]
```diff
@@ -224,6 +224,10 @@ func (s Specs) getJobNameCount() map[Name]int {
 
 type Name string
 
+func (n Name) GetConsoleURN(tnnt tenant.Tenant) string {
```
[the remark this answers is not in the document]
Comment (irainia) [reply core/job/spec.go:227]: by the way this is in context of documentation, as I couldn't find it in the `main` and this PR (at the time of writing)

File: core/resource/service/resource_service_test.go [hunk 3]
```diff
@@ -690,7 +698,9 @@ func TestResourceService(t *testing.T) {
 			mgr.On("GetURN", mock.Anything).Return(urn, nil)
 
 			eventHandler := newEventHandler(t)
-			rscService := service.NewResourceService(logger, repo, nil, mgr, eventHandler, nil)
+			alertmanager := new(mockAlertManager)
```
[the remark this answers is not in the document]
Comment (Mryashbhardwaj) [reply core/resource/service/resource_service_test.go:701]: sure

File: core/job/job.go [hunk 4]
```diff
@@ -121,6 +126,25 @@ func (j *Job) GetStaticUpstreamsToResolve() ([]*Upstream, error) {
...
+func (j ChangeType) String() string {
+	return string(j)
+}
+
+type AlertAttrs struct {
+	Name      Name
+	URN       string
+	Tenant    tenant.Tenant
+	EventTime time.Time
+	EventType ChangeType
```
[the remark this answers is not in the document]
Comment (Mryashbhardwaj) [reply core/job/job.go:145]: sure

**Pull request merged**